modules/apps/transfer/types/transfer_authorization.go

```diff
@@ -166,39 +164,22 @@ func validateMemo(ctx sdk.Context, memo string, allowedPacketDataList []string)
 	}
 
 	// if allowedPacketDataList has only 1 element and it equals AllowAllPacketDataKeys
-	// then accept all the packet data keys
+	// then accept all the memo strings
 	if len(allowedPacketDataList) == 1 && allowedPacketDataList[0] == AllowAllPacketDataKeys {
 		return nil
 	}
 
-	jsonObject := make(map[string]interface{})
-	err := json.Unmarshal([]byte(memo), &jsonObject)
-	if err != nil {
-		return err
-	}
-
 	gasCostPerIteration := ctx.KVGasConfig().IterNextCostFlat
 
-	for _, key := range allowedPacketDataList {
+	for _, allowedMemo := range allowedPacketDataList {
 		ctx.GasMeter().ConsumeGas(gasCostPerIteration, "transfer authorization")
 
-		_, ok := jsonObject[key]
-		if ok {
-			delete(jsonObject, key)
+		if strings.TrimSpace(memo) == strings.TrimSpace(allowedMemo) {
```

Comment:
```suggestion
		if strings.TrimSpace(memo) != strings.TrimSpace(allowedMemo) {
```
can we reverse the condition for ease of reading? We should just return an error if they are not equal rather than vice versa

Reply:
But that wouldn't be correct, right? If we iterate over the list of allowed memos and the first one doesn't match the memo field, but it matches the second, then we would return an error too early, wouldn't we?

Reply:
Ah makes sense. Instead, maybe we can use `slices.ContainsFunc` then? And do the gas consumption in there too? (For better readability)

Reply:
Yes, good idea! I will change it.